Core/SiteRating.swift

```diff
@@ -71,3 +71,30 @@ public struct SiteRating {
         return SiteGrade.grade(fromScore: siteScore)
     }
 }
+
+
+public class SiteRatingCache {
+    
+    public static let shared = SiteRatingCache()
+    
+    private var cachedScores = [String: Int]()
+
+    
+    /**
+     Registers a site score for in-memory caching. Only updates the score
+     in the cache if the new score is higher.
+     - returns: The score maintaned in the cache
+     */
+    func register(domain: String, score current: Int) -> Int {
```

Comment:
I agree. I wonder if "update" works better as it's not uncommon to have the "result" of an update returned. In truth I'm struggling to name it. Let me know if you have any ideas! If not, it's  probably a sign that the method is doing too much and I should split it out